.github/workflows/action-run-semgrep.yaml

```diff
@@ -0,0 +1,216 @@
+name: Semgrep CI
+on:
+  workflow_call:
+    secrets:
+      SARIF_TO_GH_COMMENT_APP_ID:
+        required: true
+        description: >
+          The ID of the GitHub application that exports SARIF results as pull request
+          annotations.
+      SARIF_TO_GH_COMMENT_APP_INSTALL_ID:
+        required: true
+        description: >
+          The repository or organization's installation ID of the GitHub application
+          exporting SARIF results as pull request annotations.
+      SARIF_TO_GH_COMMENT_APP_KEY:
+        required: true
+        description: >
+          The private key of the GitHub application exporting SARIF results
+          as pull request annotations.
+    inputs:
+      check_name:
+        type: string
+        default: ""
+        description: |
+          The name of the check, defaults the tool name from the SARIF file
+          (usually "Semgrep OSS").
+          `check_name` should be set to a different name if there are multiple
+          Semgrep workflows running inside a same pull request.
+
+      # Note: anchors are not supported by GH workflows thus we need to copy-paste
+      # the inputs.
+      # See: https://github.com/actions/runner/issues/1182
+      use_gitignore:
+        type: boolean
+        default: true
+        description: |
+          Whether to respect the .gitignore file or not. If set to 'false', then
+          only the '.semgrepignore' file will be used.
```

Comment:
In file `.semgrepignore` you have:
```
:include .gitignore
```
I might be wrong, but it looks for me that no matter if you set this flag, the `.gitignore` will always be included 🤔

Reply:
> I might be wrong, but it looks for me that no matter if you set this flag, the `.gitignore` will always be included 🤔

One important note that may not be obvious: this repository's `.semgrepignore` file only affects how the Git repository `saleor/semgrep-rules` is scanned, not how other git repositories are scanned (they will be invoking this repo's `.github/workflows/action-run-semgrep.yaml`).

Putting either `true` or `false` depend on the use case overall, by default both `.semgrepignore` and `.gitignore` are respected[^1] (i.e., if `.semgrepignore` is present, it still looks at `.gitignore` by default) but sometimes the `.gitignore` file may be excluding files that we would wish to scan (e.g., a repo that uses/used `git add -f [...]`)


It makes me wonder whether we shouldn't make `use_gitignore: false` the default to forcefully always use either the default semgrep ignore rules[^2] or custom ones (presence of a `.semgrepignore`) as people may be abusing `git add -f [...]` which could lead to critical files to not be scanned by default.<br/>
The only downside is that it it would cause Semgrep to potentially scan CI artifacts that are already excluded via `.gitignore` which could potentially increase the scan duration, but it shouldn't happen during `semgrep ci`[^3] as it only scans the git diffs (in contrast to `semgrep scan` command which does a full scan without git-diff awareness).

Overall, in my opinion, being explicit is better than implicit and thus disabling `.gitignore` by default could be a good thing. Any thoughts?

We could also consider dropping that workflow input in favor of always passing `--no-git-ignore`[^4] which would very slightly simplify the workflow, and instead users could put `:include .gitignore` in `.semgrepignore` but it could potentially cause to increase complexity on the user's end in return which may not be worth.

> In file `.semgrepignore` you have:
> 
> ```
> :include .gitignore
> ```

`:include .gitignore` is redundant, thus I will remove it. Thanks!


[^1]: https://github.com/semgrep/semgrep/blob/6b0a7e16e01906b0cdaae5c6adb9a2d02fa4e07d/cli/src/semgrep/target_manager.py#L204-L236
[^2]: https://github.com/semgrep/semgrep/blob/develop/cli/src/semgrep/templates/.semgrepignore
[^3]: https://github.com/saleor/semgrep-rules/blob/31b0c3855a3127b59c83e39af7739b80c1c5b745/.github/workflows/action-run-semgrep.yaml#L169
[^4]: https://github.com/saleor/semgrep-rules/blob/31b0c3855a3127b59c83e39af7739b80c1c5b745/.github/workflows/action-run-semgrep.yaml#L145

Reply:
As I'm not receiving feedback, I went ahead and disabled `.gitignore` in favor of being explicit in `.semgrepignore` to prevent users from accidentally and unknowingly ignoring files as those forcefully added files can be hard to notice in PRs: 0557b56c1cb17060f0b030d67ecf60508eb422fb